### ISSUE TITLE
services: Port main services list to Table component 

### ISSUE THREAD

**Issue by KKoukiou** [cockpit-project/cockpit#15797]
Came up in the UI review.

* keep name and description in one column which will wrap in smaller screen
* disable PF4 defaut mobile mode
*  ensure that table columns with long texts are wrapped or truncated
* Intergrate the filtering

**Comment by garrett** [comment]
Sorting by state would have to be something like:
1. Enabled
2. Disabled
3. Others (in some order)

Sorting the state in alphabetical order doesn't make any sense, as the names are arbitrary and would even differ between languages.

Therefore, sorting probably doesn't make much sense, as we'd want to sort by the name of the service (which is what we currently do by default anyway...) or by state, in a meaningful way. Sorting by description makes no sense. Sorting by error also doesn't make sense, as we already basically sort by errors first, then alphabetical name.

---

**TL;DR**: We don't need custom sorting. (Filtering, on the other hand, is useful, and we should definitely keep it and probably enhance it.)

**Comment by garrett** [comment]
Additionally:

- de-duplicate the name and the description (as it seems systemd sometimes returns descriptions as the same string as the name of the service)

**Comment by KKoukiou** [comment]
> Additionally:
> 
>     * de-duplicate the name and the description (as it seems systemd sometimes returns descriptions as the same string as the name of the service)

https://github.com/cockpit-project/cockpit/issues/15798
